src/Teams/Data/Repositories/QueuedProgramRepository.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Teams.Domain;
+using Teams.Models;
+
+namespace Teams.Data.Repositories
+{
+    public class QueuedProgramRepository : IQueuedProgramRepository
+    {
+        private ApplicationDbContext _db;
+        public QueuedProgramRepository(ApplicationDbContext db)
+        {
+            _db = db;
+        }
+        public void Add(ProgramCodeQuestion question, string text)
+        {
+            var model = new QueuedProgram()
+            {
+                QuestionId = question.Id,
+                Program = text,
+                Status = 0
+            };
+            _db.QueuedPrograms.Add(model);
+        }
+        public void Save()
+        {
+            _db.SaveChanges();
+        }
```

Comment:
Could you remove the method?

Reply:
But where should I save changes then? I added this method to implement an ability to save multiple entities in one transaction.

Reply:
Use dbContext directly in the controller

Reply:
I tried it, but the question wasn't saved to the db. I think I can save data only in the repository when I've added this data, otherwise it is lost, isn't it?

Reply:
No, that's not quite correct. While the same dbContext is used everywhere, it shouldn't be an issue.

Reply:
Oh, maybe I've understood it wrong. Firstly I added the context to the controller via constructor. And it didn't work. But if I use dbContext directly from the repository in the controller, but with no additional methods, all works fine. Did you mean such behaviour?

Reply:
No, it should be added via constructor everywhere. The DI container will pass the same copy of the dbContext to the repository and to the controller. If it doesn't work for you, could you submit your code with the attempt to do that?

Reply:
Ok, I'll do that.